subprojects/platform-play/src/main/java/org/gradle/language/twirl/internal/DefaultTwirlTemplateFormat.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.twirl.internal;
+
+import org.gradle.language.twirl.TwirlTemplateFormat;
+
+import java.io.Serializable;
+import java.util.Collection;
+
+public class DefaultTwirlTemplateFormat implements TwirlTemplateFormat, Serializable {
+    private final String extension;
+    private final String formatType;
+    private final Collection<String> imports;
+
+    public DefaultTwirlTemplateFormat(String extension, String formatType, Collection<String> imports) {
+        this.extension = "." + extension;
```

Comment:
The javadoc for `TwirlTemplateFormat` states that `getExtension()` does _not_ include the leading dot. I think this would be better.
We should also verify the supplied value does not begin with '.'